src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -717,7 +717,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     current.OriginalDefinition.AddUseSiteDiagnostics(useSiteDiagnostics)
                 End If
 
-                If current = superType Then
+                If current.IsSameType(superType, TypeCompareKind.IgnoreTupleNames) Then
```

Comment:
Yes, I think this is what we want. Modifiers are not taken into account by the language. They are interesting only if we are comparing member "signatures" using run-time semantics.